File: tests/fixtures/sample-multilang-app/sample.c
/* Sample C file for Zaria multi-language fixture. */
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_NAME 64
#define MAX_RECORDS 1024

typedef struct {
    int id;
    char name[MAX_NAME];
    double value;
} Record;

static int record_count = 0;
static Record records[MAX_RECORDS];

int add_record(int id, const char *name, double value) {
    if (record_count >= MAX_RECORDS) return -1;
    records[record_count].id = id;
    strncpy(records[record_count].name, name, MAX_NAME - 1);
    records[record_count].name[MAX_NAME - 1] = '\0';
    records[record_count].value = value;
    record_count++;
    return 0;
}

void trim_names(void) {
    for (int i = 0; i < record_count; i++) {
        char *p = records[i].name;
        size_t len = strlen(p);
        while (len > 0 && p[len - 1] == ' ') {
            p[--len] = '\0';
        }
    }
}

char *load_file(const char *path) {
    FILE *f = fopen(path, "rb");
    if (!f) return NULL;
    fseek(f, 0, SEEK_END);
    long sz = ftell(f);
    rewind(f);
    char *buf = malloc((size_t)sz + 1);
    if (!buf) { fclose(f); return NULL; }
    fread(buf, 1, (size_t)sz, f);
    buf[sz] = '\0';
    fclose(f);
    return buf;
}

int main(int argc, char **argv) {
    if (argc < 2) {
        fprintf(stderr, "usage: sample <path>\n");
        return 1;
    }
    char *content = load_file(argv[1]);
    if (!content) {
        perror("load_file");
        return 1;
    }
    printf("%.80s\n", content);
    free(content);
    return 0;
}

File: tests/fixtures/sample-multilang-app/sample.py
"""Sample Python module for Zaria multi-language fixture."""

import os
import sys
from pathlib import Path
from collections import defaultdict


class DataProcessor:
    """Process a list of records."""

    def __init__(self, records: list) -> None:
        self.records = records
        self._cache: dict = defaultdict(list)

    def process(self) -> list:
        results = []
        for record in self.records:
            results.append(self._transform(record))
        return results

    def _transform(self, record: dict) -> dict:
        return {k: v.strip() if isinstance(v, str) else v for k, v in record.items()}


def load_file(path: str) -> str:
    """Read a file and return its contents."""
    with open(path, encoding="utf-8") as fh:
        return fh.read()


def write_file(path: str, content: str) -> None:
    """Write content to a file."""
    with open(path, "w", encoding="utf-8") as fh:
        fh.write(content)


async def fetch_data(url: str) -> bytes:
    """Async placeholder for HTTP fetch."""
    raise NotImplementedError(url)


def main() -> int:
    """Entry point."""
    if len(sys.argv) < 2:
        print("Usage: sample.py <path>", file=sys.stderr)
        return 1
    data = load_file(sys.argv[1])
    print(data[:80])
    return 0


if __name__ == "__main__":
    sys.exit(main())
